ee_tests/src/specs/quickstart_cheterminal.spec.ts

```diff
@@ -0,0 +1,121 @@
+import { SpaceChePage } from './page_objects/space_che.page';
+import { SpaceCheWorkspacePage } from './page_objects/space_cheworkspace.page';
+import { MainDashboardPage } from './page_objects/main_dashboard.page';
+
+let globalSpaceName: string;
+
+/* Tests to verify the build pipeline */
+
+describe('Creating new quickstart in OSIO', () => {
+  let dashboardPage: MainDashboardPage;
+
+  beforeEach(async () => {
+    await support.desktopTestSetup();
+    let login = new support.LoginInteraction();
+    dashboardPage = await login.run();
+
+    let userProfilePage = await dashboardPage.gotoUserProfile();
+    support.debug(">>> Go to user's Profile Page - OK");
+    support.debug('>>> Go to Edit Profile Page');
+    let editProfilePage = await userProfilePage.gotoEditProfile();
+    support.debug('>>> Go to Edit Profile Page - OK');
+    support.debug('>>> Go to Reset Env Page');
+    let cleanupEnvPage = await editProfilePage.gotoResetEnvironment();
+    support.debug('>>> Go to Reset Env Page - OK');
+
+    await cleanupEnvPage.cleanup(browser.params.login.user);
+
+  });
+
+  afterEach( async () => {
+    await browser.sleep(support.DEFAULT_WAIT);
+
+    support.writeScreenshot('target/screenshots/che_final_' + globalSpaceName + '.png');
+    support.info('\n ============ End of test reached ============ \n');
+    // support.info('\n ============ End of test reached, logging out ============ \n');
+    /* Logout is causing random failures on prod-preview - possibky due to navigating browser windows? */
+    // await dashboardPage.logout();
+  });
+
+  /* Simple test - accept all defaults for new quickstarts */
+
+    // tslint:disable:max-line-length
+  it('Create a new space, new ' + browser.params.quickstart.name + ' quickstart, create Che workspace, run maven in the workspace', async () => {
+    let quickstart = new Quickstart(browser.params.quickstart.name);
+    let spaceName = support.newSpaceName();
+    globalSpaceName = spaceName;
+    let spaceDashboardPage = await dashboardPage.createNewSpace(spaceName);
+
+    let wizard = await spaceDashboardPage.addToSpace();
+
+    support.info('Creating quickstart: ' + quickstart.name);
+    await wizard.newQuickstartProject({ project: quickstart.name });
+    await spaceDashboardPage.ready();
+
+    /* This statement does not reliably wait for the modal dialog to disappear:
+       await browser.wait(until.not(until.visibilityOf(spaceDashboardPage.modalFade)), support.LONGEST_WAIT);
+
+       The above statement fails with this error: Failed: unknown error: Element <a id="spacehome-pipelines-title"
+       href="/username/spaceName/create/pipelines">...</a> is not clickable at point (725, 667). Other element would
+       receive the click: <modal-container class="modal fade" role="dialog" tabindex="-1" style="display:
+       block;">...</modal-container>
+
+       The only reliable way to avoid this is a sleep statement: await browser.sleep(5000); */
+    await browser.sleep(5000);
```

Comment:
Making explicit waits for fixed timeout is harmful for two reasons

- you slow down tests by predefined timeout each and every time, even if the condition is satisfied 10x faster
- it can be flaky, as it might happen that condition will appear a bit later than the timeout you defined

Is it really the only way to handle this case?

Reply:
As it turns out - that code is no longer needed - it might have been needed in the past due to the UI being unstable. Either way, the test runs cleanly now without it. Thx!